src/jit/emitxarch.cpp

```diff
@@ -2122,7 +2226,7 @@ UNATIVE_OFFSET emitter::emitInsSizeAM(instrDesc* id, code_t code)
         size += emitGetRexPrefixSize(ins);
     }
     else if (IsExtendedReg(reg, EA_PTRSIZE) || IsExtendedReg(rgx, EA_PTRSIZE) ||
-             ((ins != INS_call) && IsExtendedReg(id->idReg1(), attrSize)))
+             ((ins != INS_call) && (IsExtendedReg(id->idReg1(), attrSize) || IsExtendedReg(id->idReg2(), attrSize))))
```

Comment:
We should never be in a case where `idReg3` or `idReg4` are set for the addressing mode instructions (AM, SV, or CV). The instruction descriptor also doesn't allow this since those slots are used for the index and base registers in those cases.